jujugui/static/gui/src/app/app.js

```diff
@@ -720,10 +720,16 @@ YUI.add('juju-gui', function(Y) {
       @method _renderLogout
     */
     _renderLogout: function() {
-      ReactDOM.render(
-        <window.juju.components.Logout
-          logout={this.logout.bind(this)} />,
-        document.getElementById('profile-link-container'));
+      var node = document.getElementById('profile-link-container');
+      if(this.state.getState('current', 'sectionC', 'metadata')) {
+        // If the charmbrowser is open then don't show the logout link.
+        ReactDOM.unmountComponentAtNode(node);
```

Comment:
So I think that instead of unmounting you need to trigger an animation on the component because it is pretty janky when clicking on search or midpoint.

```JavaScript
    _renderLogout: function() {
      var node = document.getElementById('profile-link-container');
      // If the charmbrowser is open then don't show the logout link.
      var open = this.state.getState('current', 'sectionC', 'metadata') ? false : true;
      ReactDOM.render(
        <window.juju.components.Logout
          open={open}
          logout={this.logout.bind(this)} />,
        node);
    },
```

Then the component will need to be updated to handle this new prop.